src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -135,36 +139,48 @@ class NativeAppRunProcessor(NativeAppManager, NativeAppCommandProcessor):
     def __init__(self, project_definition: NativeApp, project_root: Path):
         super().__init__(project_definition, project_root)
 
-    def _execute_sql_script(self, sql_script_path):
+    def _execute_sql_script(
+        self, script_content: str, database_name: Optional[str] = None
+    ):
         """
-        Executing the SQL script in the provided file path after expanding template variables.
+        Executing the provided SQL script content.
         This assumes that a relevant warehouse is already active.
-        Consequently, "use database" will be executed first if it is set in definition file or in the current connection.
+        If database_name is passed in, it will be used first.
         """
-        with open(sql_script_path) as f:
-            sql_script = f.read()
-
         try:
-            if self._conn.database:
-                self._execute_query(f"use database {self._conn.database}")
-            sql_script = snowflake_sql_jinja_render(content=sql_script)
-            self._execute_queries(sql_script)
+            if database_name:
```

Comment:
nit: `is not None` is the typical way to check this, since we shouldn't make a habit of interpreting what it means for a string to be truthy.